doc/source/security/data-at-rest-encryption.rst

```diff
@@ -0,0 +1,390 @@
+.. _data_at_rest_encryption:
+
+===============================================================================
+Data at Rest Encryption
+===============================================================================
+
+.. contents::
+   :local:
+
+|Percona Server| enables data at rest encryption of the InnoDB (file-per-table) tablespace by encrypting the physical database files. The data is automatically encrypted prior to writing to storage and automatically decrypted when read.
+If unauthorized users access the data files, they cannot read the contents. 
+
+.. rubric:: Architecture
+
+The data at rest encryption uses a two-tier architecture with the following components:
+
+.. tabularcolumns:: |p{5cm}|p{11cm}|
+
+.. list-table::
+   :header-rows: 1
+  
+   * - Type
+     - Description
+   * - Master key
+     - The Master key is used to encrypt or decrypt the tablespace keys.
+   * - Tablespace key for each tablespace
+     - The tablespace key encrypts the data pages and is written in the tablespace header. Rebuild the table to change the tablespace key.
+
+When the server must access the data, the master key decrypts the tablespace key, the tablespace is decrypted and available for read or write operations.
+
+The two separate keys architecture allows the master key to be rotated in a
+minimal operation. During the master key rotation, each tablespace key is
+re-encrypted with the new master key. Only the first page of the tablespace file
+(.ibd) is read and written during the rotation. An encrypted page is decrypted
+at the I/O layer, added to the buffer pool, and used to read and write the data.
+A buffer pool page is not encrypted. The I/O layer encrypts the page before the
+page is flushed to disk.
+
+An encryption key is required to encrypt or decrypt the tablespace. The key is stored in the keyring plugin.
```

Comment:
That's not entirely correct. The key to decrypt tablespace, as you wrote above is stored in the tablespace's header. The Master Key is what's there in keyring plugin.

Reply:
Fixed.